### PR TITLE
BUG: Don't transpose variables of one dimension

### PR DESCRIPTION
**Pull request by MaximilianR**
Resolves https://github.com/pydata/xarray/issues/745

A bit of a 'convenient hack'. Let me know if you think there's a better way to do this

### CODE REVIEW

File: xarray/core/variable.py [hunk 1]
```diff
@@ -647,6 +647,9 @@ def transpose(self, *dims):
         """
         if len(dims) == 0:
             dims = self.dims[::-1]
+        # no need to transpose if only one dimension
+        if len(dims) == 1:
+            return self
```
Comment (shoyer) [inline xarray/core/variable.py]: It's OK to keep the data the same, but we shouldn't return the same `Variable` object. I would use `self.copy(deep=False)` instead. Also, let's check *after* we verify that `dims` is valid (e.g., move the conditional one line below). Probably a little safer to use `if self.ndim < 2` as the check instead, too.

**Comment by MaximilianR** [comment]
Great, @shoyer, updated

**Comment by MaximilianR** [comment]
Failing on this: https://github.com/pydata/pandas/issues/12235, will push again when resolved

**Comment by MaximilianR** [comment]
@shoyer green

**Pull request merged**